server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -286,9 +293,12 @@
      * the chance of ops based recoveries for indices with soft-deletes disabled.
      * This setting will be ignored if soft-deletes is used in peer recoveries (default in 7.4).
      **/
+    private static final ByteSizeValue DEFAULT_TRANSLOG_RETENTION_SIZE = new ByteSizeValue(512, ByteSizeUnit.MB);
+
     public static final Setting<ByteSizeValue> INDEX_TRANSLOG_RETENTION_SIZE_SETTING =
         Setting.byteSizeSetting("index.translog.retention.size",
-            settings -> shouldDisableTranslogRetention(settings) ? "-1" : "512MB",
+            settings -> shouldDisableTranslogRetention(settings) &&
```

Comment:
This seems off. Why not directly use translog retention setting value in order to decide how much translog to prune?

Reply:
This logic is modified to keep the existing behavior as is - and change based on the translog pruning setting.